### PR TITLE
feat(derive): InMemory Blob Provider

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Provides a concrete implementation for the `BlobProvider` using an in-memory blob store.

### CODE REVIEW

**Comment by refcell** [comment]
* **#216** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/216?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @refcell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by clabby — APPROVED** [review]
lgtm, 1 q

File: crates/derive/src/traits/blobs.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        };
+        let locked = self.0.lock();
+        let blobs_for_block =
+            locked.blocks_to_blob.get(&block_ref.hash).ok_or_else(|| err(block_ref))?;
+        let mut blobs = Vec::new();
+        for _blob_hash in blob_hashes {
+            for blob in blobs_for_block {
+                blobs.push(*blob);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (refcell) [reply crates/derive/src/traits/blobs.rs:74]: Duh. Of course you're right. We need to push only if the blob hash is verified.

[the remark this answers is not in the document]
Comment (refcell) [reply crates/derive/src/traits/blobs.rs:74]: Do you actually know if the blob hash is just the keccak256 hash of the raw blob data?

[the remark this answers is not in the document]
Comment (clabby) [reply crates/derive/src/traits/blobs.rs:74]: The versioned hash is `BLOB_VERSION_BYTE ++ sha256(kzg_commitment)[1..]`, per https://hackmd.io/@protolambda/ethdenver_data_blob_transactions#Helpers

**Pull request closed without merging**